### PR TITLE
https image urls if accessed over SSL

### PR DESCRIPTION
**Pull request by sovietspaceship2**
Tentative fix for #8372. I'm not familiar with this codebase so I don't know the rationale behind the current code nor whether this change would be enough, but hopefully will point to the right direction.

### CODE REVIEW

**Review by kirrg001 — CHANGES_REQUESTED** [review]
Thanks for your PR!

Could you please add a test case to proof your case is not working?

Probably in `core/test/unit/utils/url_spec.js`.

**Comment by sovietspaceship** [comment]
Our blog had optional https but http by default, so it was configured with an http base url. The https version of the rss feed was returning http links for images as the current code only checks the image path for https, ignoring that the actual connection may be https. This PR fixes it.

File: core/server/utils/url.js [hunk 1]
```diff
@@ -259,7 +259,7 @@ function urlFor(context, data, absolute) {
             urlPath = data.image;
             imagePathRe = new RegExp('^' + getSubdir() + '/' + STATIC_IMAGE_URL_PREFIX);
             absolute = imagePathRe.test(data.image) ? absolute : false;
-            secure = data.image.secure;
+            secure = data.image.secure || secure;
```
Comment (kirrg001) [inline core/server/utils/url.js]: Sorry for late review!

I just double checked this line of code. `data.image.secure` is never a case. We expect `image` to be the url path. If you remove the whole line, it should work as expected.
Thanks for adding the test 👍

Comment (sovietspaceship) [inline core/server/utils/url.js]: You're right, data.image is a string.
Done.

**Review by kirrg001 — APPROVED** [review]
👍 Thank you!

**Pull request merged**